cli/command/image/load.go

```diff
@@ -63,12 +69,21 @@ func runLoad(ctx context.Context, dockerCli command.Cli, opts loadOptions) error
 		return errors.Errorf("requested load from stdin, but stdin is empty")
 	}
 
-	var loadOpts image.LoadOptions
+	var options image.LoadOptions
 	if opts.quiet || !dockerCli.Out().IsTerminal() {
-		loadOpts.Quiet = true
+		options.Quiet = true
 	}
 
-	response, err := dockerCli.Client().ImageLoad(ctx, input, loadOpts)
+	if opts.platform != "" {
+		p, err := platforms.Parse(opts.platform)
+		if err != nil {
+			_, _ = fmt.Fprintf(dockerCli.Err(), "Invalid platform %s", opts.platform)
+			return err
```

Comment:
I just realised that this isn't correct; it tries to both _print_ something as well as returning an error. The printed message is without a newline. But also, because the error is returned, both the printed message and the error get printed;


```bash
docker image save -o image-amd64.tar --platform in:valid/foo:bar  alpine:latest
Invalid platform in:valid/foo:bar"in:valid" is an invalid OS component of "in:valid/foo:bar": OSAndVersion specifier component must match "^([A-Za-z0-9_-]+)(?:\\(([A-Za-z0-9_.-]*)\\))?$": invalid argument
```

I wondered if this was perhaps done for cases where no output file is specified (so the tar output would be redirected), but the same happens in that case;


```bash
docker image save --platform in:valid/foo:bar  alpine:latest > file.tar
Invalid platform in:valid/foo:bar"in:valid" is an invalid OS component of "in:valid/foo:bar": OSAndVersion specifier component must match "^([A-Za-z0-9_-]+)(?:\\(([A-Za-z0-9_.-]*)\\))?$": invalid argument
```


So I think this should just wrap the error and return it.

Reply:
Updated these to just return the error